cmd/controller/ctrl.go

```diff
@@ -163,11 +188,12 @@ func deleteDR(name string, c *Controller) {
 }
 
 // deactivateStaleDiskResource deactivates the stale entry from etcd
-func deactivateStaleDiskResource(c *Controller, detected v1.BlockDeviceInfo, listDR *apis.DiskList) {
+func deactivateStaleDiskResource(c *Controller, devices []*udev.Udevice, listDR *apis.DiskList) {
 	for _, item := range listDR.Items {
 		var uuid string
-		for _, blk := range detected.Blockdevices {
-			uuid = getUid(blk)
+		for i := range devices {
```

Comment:
You can use `for _ ,device := range devices {`
instead of
`for i := range devices {
		device := devices[i]`